sources/api/pluto/src/api.rs

```diff
@@ -1,76 +1,101 @@
+    pub kubernetes: Option<Kubernetes>,
+}
 
-        serde_json::from_str(&response_body).context(SettingsJsonSnafu)
-    }
+#[derive(Deserialize)]
+struct SettingsView {
+    pub settings: View,
+}
 
-    /// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        let settings = get_settings().await?;
-        Ok(AwsK8sInfo {
-            region: settings.aws.and_then(|a| a.region).map(|s| s.into()),
-            cluster_name: settings
-                .kubernetes
-                .as_ref()
-                .and_then(|k| k.cluster_name.clone())
-                .map(|s| s.into()),
-            cluster_dns_ip: settings.kubernetes.and_then(|k| k.cluster_dns_ip),
-        })
-    }
+#[derive(Debug, Snafu)]
+pub(crate) enum Error {
+    #[snafu(display("Failed to call apiclient: {}", source))]
+    CommandFailure { source: std::io::Error },
+    #[snafu(display("APIClient execution failed: {}", reason))]
+    ExecutionFailure { reason: String },
+    #[snafu(display("Deserialization of configuration file failed: {}", source))]
+    Deserialize {
+        #[snafu(source(from(serde_json::Error, Box::new)))]
+        source: Box<serde_json::Error>,
+    },
 }
 
-/// This dummy code is compiled when the `sources` workspace is being compiled for non `aws-k8s-*`
-/// variants.
-#[cfg(not(variant_family = "aws-k8s"))]
-mod inner {
-    use super::*;
-    use snafu::Snafu;
+pub(crate) async fn client_command<I, S>(args: I) -> Result<Vec<u8>>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let result = Command::new("/usr/bin/apiclient")
+        .args(args)
+        .output()
+        .await
+        .context(CommandFailureSnafu)?;
 
-    #[derive(Debug, Snafu)]
-    pub(crate) enum Error {
-        #[snafu(display(
-            "The get_aws_k8s_info function is only compatible with aws-k8s variants"
-        ))]
-        WrongVariant,
-    }
+    ensure!(
+        result.status.success(),
+        ExecutionFailureSnafu {
+            reason: String::from_utf8_lossy(&result.stderr)
+        }
+    );
+
+    Ok(result.stdout)
+}
 
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        WrongVariantSnafu.fail()
-    }
+/// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
+pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
+    let view_str = client_command(&[
+        "get",
+        "settings.aws.region",
+        "settings.kubernetes.cluster-name",
+        "settings.kubernetes.cluster-dns-ip",
+    ])
+    .await?;
+    let view: SettingsView =
+        serde_json::from_slice(view_str.as_slice()).context(DeserializeSnafu)?;
+    Ok(AwsK8sInfo {
+        region: view.settings.aws.and_then(|a| a.region),
+        cluster_name: view
+            .settings
+            .kubernetes
+            .as_ref()
+            .and_then(|k| k.cluster_name.clone()),
```

Comment:
Isn't the cluster_name eventually required? Can it be required here instead of being an `Option`?

Reply:
I'll need to validate that it doesn't break anything to require it and not use an Option. Generally I followed our convention that #[model] does

Reply:
While I think you are probably right, talked offline with ben to see if there are cases where it isn't set and don't think there is need to really to take the risk of being surprised by it if it fails